eth/tracers/api.go

```diff
@@ -638,12 +641,19 @@ func (api *API) traceBlock(ctx context.Context, block *types.Block, config *Trac
 			}
 		}()
 	}
+
 	// Feed the transactions into the tracers and return
 	var failed error
 	blockCtx := core.NewEVMBlockContext(block.Header(), api.chainContext(ctx), nil)
 	for i, tx := range txs {
 		// Send the trace task over for execution
-		jobs <- &txTraceTask{statedb: statedb.Copy(), index: i}
+		task := &txTraceTask{statedb: statedb.Copy(), index: i}
+		select {
+		case jobs <- task:
+		case <-ctx.Done():
+			failed = ctx.Err()
+			break
```

Comment:
Classic mistake here, `break` here will terminate the `select` instead of enclosing loop.